src/openrct2/peep/Guest.cpp

```diff
@@ -3596,10 +3596,20 @@ void rct_peep::UpdateRideAdvanceThroughEntrance()
             peep_update_ride_leave_entrance_maze(this, ride, entranceLocation);
             return;
         }
-        Guard::Assert(ride->type == RIDE_TYPE_SPIRAL_SLIDE);
+        else if (ride->type == RIDE_TYPE_SPIRAL_SLIDE)
+        {
+            peep_update_ride_leave_entrance_spiral_slide(this, ride, entranceLocation);
+            return;
+        }
+        else
+        {
+            // If the ride type was changed guests will become stuck.
+            auto intent = Intent(WC_RIDE);
+            intent.putExtra(INTENT_EXTRA_RIDE_ID, current_ride);
+            context_open_intent(&intent);
 
-        peep_update_ride_leave_entrance_spiral_slide(this, ride, entranceLocation);
-        return;
+            context_show_error(STR_WARNING, STR_GUESTS_GETTING_STUCK_ON_RIDE);
```

Comment:
The previous code returned at this point we now fall through into the code that assumes we have vehicles on the ride. If we don't have vehicles on the ride then this will cause a nullptr deref. Is this what we want. I can't remember off the top of my head if rides always have at least one vehicle.

Reply:
You are right, its an out of bound but I was not able to actually trigger it. I'll add the return statement back. As for the spam, perhaps its better to do this on the host/local game only?